projects/ui/src/Components/Apps/Details/Modals/ConfirmDeleteOAuthModal.tsx

```diff
@@ -0,0 +1,61 @@
+import { Box, CloseButton, Flex } from "@mantine/core";
+import { FormEvent } from "react";
+import toast from "react-hot-toast";
+import { di } from "react-magnetic-di";
+import { useDeleteOAuthMutation } from "../../../../Apis/gg_hooks";
+import { FormModalStyles } from "../../../../Styles/shared/FormModalStyles";
+import { Button } from "../../../Common/Button";
+
+const ConfirmDeleteOAuthModal = ({
+  credentialId,
+  appId,
+  open,
+  onClose,
+}: {
+  credentialId: string;
+  appId: string;
+  open: boolean;
+  onClose: () => void;
+}) => {
+  di(useDeleteOAuthMutation);
+  const { trigger: deleteOAuth } = useDeleteOAuthMutation(appId);
+  const onConfirm = async (e?: FormEvent) => {
```

Comment:
Similar to above, I wonder that this isn't a standardized call as well (preventDefault, action, close)

Reply:
This whole component and a few others should be a `ConfirmDeleteModal`. As for the form event function being standardized/abstracted, I don't think I'd be against it, but I'd rather revisit that later.